src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -6,19 +6,107 @@
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold;
+    private Node<K, V>[] table;
+    private int size = 0;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+    }
+
+    private class Node<K, V> {
+        K key;
+        V value;
+        Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        Node<K, V> node;
+        int index = hash(key);
```

Comment:
Maybe it would be better if the method ```hash(key)``` will be named something like ```getIndex(key)``` ? Because this method returns you the index of an element